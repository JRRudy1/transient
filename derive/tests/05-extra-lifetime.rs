//! Tests the behavior when used on structs with too many lifetime parameters
use transient_any::MakeStatic;


#[derive(Debug, Clone, PartialEq, Eq, MakeStatic)]
struct S<'a, 'b, T> {
    value1: &'a T,
    value2: &'b T,
}


fn main() {
    // this test should fail to compile
}
