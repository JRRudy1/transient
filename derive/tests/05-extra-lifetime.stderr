error: At most one lifetime parameter is allowed!
 --> tests/05-extra-lifetime.rs:6:14
  |
6 | struct S<'a, 'b, T> {
  |              ^^
